jcl/src/java.base/share/classes/java/lang/StringBuffer.java

```diff
@@ -754,8 +754,8 @@ public synchronized StringBuffer delete(int start, int end) {
 public synchronized StringBuffer deleteCharAt(int location) {
 	int currentLength = lengthInternalUnsynchronized();
 
-	if (currentLength != 0) {
-		return delete (location, location + 1);
+	if ((currentLength >= 0) && (currentLength > location)) {
```

Comment:
Using `>=` here just delays the exception; if the length is zero, there is no legal `location`.
Suggest the condition would be clearer using the pattern in `charAt()`, which becomes
```
	if ((0 <= location) && (location < currentLength)) {
```
The same goes for `StringBuilder`.

Reply:
Changed the condition to what you suggested.